b2/sync.py

```diff
@@ -761,64 +762,123 @@ def check_file_replacement(source_file, dest_file, args):
         raise CommandError('Invalid option for --compareVersions')
 
 
+def should_transfer_file(source_file, dest_file, args):
+    """
+    Decides whether to transfer the file from the source to the destination.
+    """
+    if source_file is None:
+        # No source file.  Nothing to transfer.
+        return False
+    elif dest_file is None:
+        # Source file exists, but no destination file.  Always transfer.
+        return True
+    else:
+        # Both exist.  Transfer only if the two are different.
+        return files_are_different(source_file, dest_file, args)
+
+
+def make_b2_delete_actions(source_file, dest_file, dest_folder, transferred):
```

Comment:
this function is hard to read. What does it do? The `keep = ...` line is very cryptic